libtock/ieee802154.c

```diff
@@ -268,66 +210,55 @@ int ieee802154_get_neighbor(unsigned index,
-  if (!rw.success) return tock_error_to_rcode(rw.error);
+  if (!rw.success) return tock_status_to_returncode(rw.status);
+
   syscall_return_t com = command(RADIO_DRIVER, COMMAND_ADD_NEIGHBOR, (unsigned int) addr, 0);
-  if (com.type == TOCK_SYSCALL_SUCCESS_U32 && index) {
-    // Driver adds 1 to ensure it is positive.
-    *index = (unsigned) (com.data[0] - 1);
-    return com.data[0];
-  } else if (com.type < TOCK_SYSCALL_SUCCESS) {
-    return tock_error_to_rcode(com.data[0]);
-  } else {
-    return TOCK_FAIL;
-  }
+  int ret = tock_command_return_u32_to_returncode(com, (uint32_t*) index);
+
+  // Driver adds 1 to ensure it is positive.
+  *index -= 1;
+
+  return ret;
 }
 
 int ieee802154_remove_neighbor(unsigned index) {
   syscall_return_t com = command(RADIO_DRIVER, COMMAND_REMOVE_NEIGHBOR, (unsigned int) index, 0);
-  if (com.type == TOCK_SYSCALL_SUCCESS) {
-    return TOCK_SUCCESS;
-  } else if (com.type > TOCK_SYSCALL_SUCCESS) {
-    return TOCK_FAIL;
-  } else {
-    return tock_error_to_rcode(com.data[0]);
-  }
+  return tock_command_return_novalue_to_returncode(com);
 }
 
-int ieee802154_max_keys(void) {
+int ieee802154_max_keys(int* keys) {
   syscall_return_t com = command(RADIO_DRIVER, COMMAND_MAX_KEYS, 0, 0);
-  // Driver adds 1 to ensure it is positive, but on error we want to return 0
-  if (com.type == TOCK_SYSCALL_SUCCESS_U32) {
-    // Driver adds 1 to the power after casting it to unsigned, so this works
-    return com.data[0] - 1;
-  } else {
-    return 0;
-  }
+  int ret = tock_command_return_u32_to_returncode(com, (uint32_t*) keys);
+
+  // Driver adds 1 to ensure it is positive.
+  *keys -= 1;
+
+  return ret;
 }
 
-int ieee802154_num_keys(void) {
+int ieee802154_num_keys(int* keys) {
   syscall_return_t com = command(RADIO_DRIVER, COMMAND_NUM_KEYS, 0, 0);
-  // Driver adds 1 to ensure it is positive, but on error we want to return 0
-  if (com.type == TOCK_SYSCALL_SUCCESS_U32) {
-    // Driver adds 1 to the power after casting it to unsigned, so this works
-    return com.data[0] - 1;
-  } else {
-    return 0;
-  }
+  int ret = tock_command_return_u32_to_returncode(com, (uint32_t*) keys);
+
+  // Driver adds 1 to ensure it is positive.
+  *keys -= 1;
+
+  return ret;
 }
 
 int ieee802154_get_key_security_level(unsigned index, security_level_t *level) {
-  if (!level) return TOCK_EINVAL;
+  if (!level) return RETURNCODE_EINVAL;
+
   syscall_return_t com = command(RADIO_DRIVER, COMMAND_GET_KEY_LEVEL, (unsigned int) index, 0);
-  if (com.type == TOCK_SYSCALL_SUCCESS_U32) {
-    // Driver adds 1 to ensure it is positive.
-    *level = (security_level_t) (com.data[0] - 1);
-    return com.data[0];
-  } else if (com.type < TOCK_SYSCALL_SUCCESS) {
-    return tock_error_to_rcode(com.data[0]);
-  } else {
-    return TOCK_FAIL;
-  }
+  int ret = tock_command_return_u32_to_returncode(com, (uint32_t*) level);
+
+  // Driver adds 1 to ensure it is positive.
+  *level -= 1;
```

Comment:
In the case of this command returning an error, this decrements an uninitialized variable. The caller shouldn't use the returned value anyway, but I think this is technically UB. I don't think it would matter in practice though.